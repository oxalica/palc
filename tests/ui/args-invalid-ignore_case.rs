use palc::{Args, ValueEnum};

#[derive(Args)]
struct Cli1 {
    #[arg(long, value_enum, ignore_case = true)]
    lower: LowerCase,
    #[arg(long, value_enum, ignore_case = true)]
    upper: UpperCase,
}

#[derive(ValueEnum)]
enum LowerCase {
    Hello,
}

#[derive(ValueEnum)]
#[value(rename_all = "UPPER")]
enum UpperCase {
    Hello,
}

fn main() {}
