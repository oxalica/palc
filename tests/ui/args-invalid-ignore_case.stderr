error[E0080]: evaluation of constant value failed
 --> tests/ui/args-invalid-ignore_case.rs:8:12
  |
8 |     upper: UpperCase,
  |            ^^^^^^^^^ evaluation panicked: `arg(ignore_case)` only supports `ValueEnum` that contains no UPPERCASE variants
